Tests/Integration/SemanticTests/external_calls.flint

```diff
@@ -7,17 +7,25 @@ contract Contract {
 Contract :: (any) {
   func someFunc() {
     
+  }
+
+  @payable
+  func payableFunc(implicit parameter: Wei) {
+
   }
   
   public init() {
     call! self.someFunc()
     call(gas: 100)! self.someFunc()
-    call(value: Wei(100))! self.someFunc()
-    call(value: Wei(100), gas: 100)! self.someFunc()
+    call(value: Wei(100), gas: 100)! self.payableFunc(parameter: Wei(1))
 
-    call(gas: 100, value: Wei(200), gas: 300)! self.someFunc() // expected-error {{'gas' hyper-parameter was already specified}}
-    call(value: Wei(100), value: Wei(200), value: Wei(300))! self.someFunc() // expected-error {{'value' hyper-parameter was already specified}}
-                                                                             // expected-error@-1 {{'value' hyper-parameter was already specified}}
+    call(gas: 100)! self.payableFunc(parameter: Wei(120)) // expected-error {{Attempting to call a payable function without specifying a value hyper-parameter}}
+    call(gas: Wei(100))! self.someFunc() // expected-error {{Gas hyper-parameter of an external call must be of type Int}}
+    call(value: Wei(100))! self.someFunc() // expected-error {{Attempting to call a non-payable function with a value hyper-parameter}}
+    call(value: 100) self.payableFunc(parameter: Wei(1)) // expected-error {{Value hyper-parameter of an external call must be of type Wei}}
+    call(gas: 100, value: Wei(200), gas: 300)! self.payableFunc(parameter: Wei(1)) // expected-error {{'gas' hyper-parameter was already specified}}
+    call(value: Wei(100), value: Wei(200), value: Wei(300))! self.payableFunc(parameter: Wei(1)) // expected-error {{'value' hyper-parameter was already specified}}
+                                                                                                 // expected-error@-1 {{'value' hyper-parameter was already specified}}
```

Comment:
Add tests for nested external calls (`call! self.someFunc(arg: call! self.someOtherFunc())`). Also tests for external calls where the argument is simply a function call (`call! self.someFunc(arg: self.someBasicFunc())`).